detection_rules/attack.py

```diff
@@ -17,43 +22,99 @@
     'Lateral Movement': 'TA0008',
     'Execution': 'TA0002',
     'Collection': 'TA0009',
-    'Exfiltration': 'TA0011',
-    'Command and Control': 'TA0010',
+    'Exfiltration': 'TA0010',
+    'Command and Control': 'TA0011',
     'Impact': 'TA0040'
 }
 TACTICS = list(TACTICS_MAP)
 PLATFORMS = ['Windows', 'macOS', 'Linux']
 
-attack = load_etc_dump('attack.json')
+
+def get_attack_file_path():
+    pattern = 'att&ck-v*.json.gz'
+    attack_file = get_etc_glob_path(pattern)
+    if len(attack_file) != 1:
+        raise FileNotFoundError(f'Missing required {pattern} file')
+    return attack_file[0]
+
+
+def load_attack_gz():
+    return json.loads(open_gzip(get_attack_file_path()))
+
+
+attack = load_attack_gz()
 
 technique_lookup = {}
+revoked = {}
 
 for item in attack["objects"]:
     if item["type"] == "attack-pattern" and item["external_references"][0]['source_name'] == 'mitre-attack':
         technique_id = item['external_references'][0]['external_id']
         technique_lookup[technique_id] = item
 
+        if item.get('revoked'):
+            revoked[technique_id] = item
+
 matrix = {tactic: [] for tactic in TACTICS}
+matrix.update(NO_TACTIC=[])
 attack_tm = 'ATT&CK\u2122'
 
 
 # Enumerate over the techniques and build the matrix back up
 for technique_id, technique in sorted(technique_lookup.items(), key=lambda kv: kv[1]['name'].lower()):
-    for platform in technique['x_mitre_platforms']:
-        if any(platform.startswith(p) for p in PLATFORMS):
-            break
-    else:
-        continue
-
-    for tactic in technique['kill_chain_phases']:
-        tactic_name = next(t for t in TACTICS if tactic['kill_chain_name'] == 'mitre-attack' and t.lower() == tactic['phase_name'].replace("-", " "))  # noqa: E501
-        matrix[tactic_name].append(technique_id)
+    kill_chain = technique.get('kill_chain_phases')
+    if kill_chain:
+        for tactic in kill_chain:
+            tactic_name = next(t for t in TACTICS if tactic['kill_chain_name'] == 'mitre-attack' and t.lower() == tactic['phase_name'].replace("-", " "))  # noqa: E501
+            matrix[tactic_name].append(technique_id)
+        else:
+            matrix['NO_TACTIC'].append(technique_id)
```

Comment:
These are unused in the current state, though may become more important with sub-techniques